### ISSUE TITLE
Seems missing `-liconv` in linux build configure

### ISSUE THREAD

**Issue by lguangyu** [tidyverse/haven#465]
HOW...
```
# wget https://cloud.r-project.org/src/contrib/haven_2.1.1.tar.gz
# untar
R CMD INSTALL ./haven
```

WHAT...
Failed load test:
```
** testing if installed package can be loaded
Error: package or namespace load failed for ‘haven’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/home/li.gua/usr/R/haven/libs/haven.so':
  /home/li.gua/usr/R/haven/libs/haven.so: undefined symbol: libiconv
Error: loading failed
Execution halted
ERROR: loading failed
```
`ldd` check confirms haven.so not linked into `libiconv.so`:
```bash
$ ldd src/haven.so 
	linux-vdso.so.1 =>  (0x00007ffd29276000)
	libz.so.1 => /home/li.gua/.local/lib/libz.so.1 (0x00002b4226b7f000)
	libR.so => not found
	libstdc++.so.6 => /shared/centos7/gcc/7.2.0/lib64/libstdc++.so.6 (0x00002b4226d9c000)
	libm.so.6 => /lib64/libm.so.6 (0x00002b422711d000)
	libgcc_s.so.1 => /shared/centos7/gcc/7.2.0/lib64/libgcc_s.so.1 (0x00002b422741f000)
	libc.so.6 => /lib64/libc.so.6 (0x00002b4227636000)
	/lib64/ld-linux-x86-64.so.2 (0x00002b42266fa000)
```

`src/Makevars`:
```
CFILES = $(wildcard *.c readstat/*.c readstat/sas/*.c readstat/spss/*.c readstat/stata/*.c)
CPPFILES = $(wildcard *.cpp)

SOURCES = $(CFILES) $(CPPFILES)

# This must be defined identically in Makevars.win
OBJECTS = $(CFILES:.c=.o) $(CPPFILES:.cpp=.o)

PKG_CFLAGS = -Ireadstat -DHAVE_ZLIB
PKG_CXXFLAGS = -Ireadstat -DHAVE_ZLIB
PKG_LIBS = -lz
```

Manually add `-liconv` to `PKG_LIBS` temporarily solve the problem.

**Comment by hadley** [comment]
It builds fine for us, so this must be something peculiar to your system.

**Comment by liutiming** [comment]
I had this issue with using tidyverse with r-essentials in conda. Using a different channel combination helped to resolve the issue. 
```
conda create --prefix directory -c conda-forge -c anaconda -c r r-essentials r-base fonts-anaconda r-cairo
```

note: r-base=4.0.3 will be installed. Changing r-base=3.6.3 will result in the same error when loading tidyverse. I suspect the conda build of tidyverse for r-base=3.6.3 probably has some issues (but solved in 4.0.3). can't find records of that tho
